serving/deploying-with-private-registry.md

```diff
@@ -0,0 +1,143 @@
+    ```
+
+A Service Account provides an identity for processes that run in a Pod. This Service Account will be used to link the build process for Knative to the Secrets you just created.
+
+1. Create a file named service-account.yaml containing the following .yaml.
+
+    ```
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: build-bot
+    secrets:
+    - name: registry-push-secret
+    imagePullSecrets:
+    - name: ibm-cr-secret
+    ```
+
+## Deploy to Knative using the credentials
+To build our application from the source on github, and push the resulting image to the IBM Container Registry, we will use the Kaniko build template.
+
+1. Install the Kaniko build template
+
+    ```
+    kubectl apply -f https://raw.githubusercontent.com/knative/build-templates/master/kaniko/kaniko.yaml
+    ```
+
+1. You need to create a service manifest which defines the service to deploy, including where the source code is and which build-template to use. Create a file named service.yaml and copy the following definition. Make sure to replace {NAMESPACE} with your own namespace you created earlier:
+
+    ```
+    apiVersion: serving.knative.dev/v1alpha1
+    kind: Service
+    metadata:
+      name: app-from-source
+      namespace: default
+    spec:
+      runLatest:
+        configuration:
+          build:
+            apiVersion: build.knative.dev/v1alpha1
+            kind: Build
+            spec:
+              serviceAccountName: build-bot
+              source:
+                git:
+                  url: https://github.com/mchmarny/simple-app.git
+                  revision: master
+              template:
+                name: kaniko
+                arguments:
+                - name: IMAGE
+                  value: registry.ng.bluemix.net/{NAMESPACE}/app-from-source:latest
+          revisionTemplate:
+            spec:
+              container:
+                image: registry.ng.bluemix.net/{NAMESPACE}/app-from-source:latest 
+                imagePullPolicy: Always
+                env:
+                - name: SIMPLE_MSG
+                  value: "Hello from the sample app!"
+    ```
+
+1. Apply the configuration using `kubectl`. Applying this service definition will enable a number of events to happen:
+- Fetch the revision specified from GitHub and build it into a container, using the Kaniko build template.
+- the latest image will be pushed to the private registry using the registry-push-secret
+- the latest image will be pulled down from the private registry using the ibm-cr-secret.
+- the service will start, and your app will be running.
+
+    ```
+    kubectl apply -f service.yaml
+    ```
+
+1. You can run `kubectl get pods --watch` to see the pods initializing.
+
+1. Once all the pods are initialized, you can see that your container image was built and pushed to the IBM Container Registry:
+
+    ```
+    ibmcloud cr image-list
+    ```
```

Comment:
I'm wondering if it might make sense for this guide to also include the info for interacting with the app from the command line? 

Similar to: https://github.com/knative/docs/blob/master/install/getting-started-knative-app.md

Reply:
I was torn --  I felt that this is a more advanced tutorial, and it would just be repeating that information in 2 places. That said, it would be nice for them to confirm everything is working by testing the app.  I'll add to this PR to include that.